### PR TITLE
 [IMP] mail: sync activities data between tabs

### PR DESCRIPTION
**Pull request by oomsveta**
Enterprise: https://github.com/odoo/enterprise/pull/38537

Task-2714099.

### CODE REVIEW

File: addons/mail/static/src/core/messaging_service.js [hunk 1]
```diff
@@ -180,6 +182,12 @@ export class Messaging {
                         this.store.activityCounter--;
                     }
                     break;
+                case "mail.activity/delete":
+                    this.activityService.delete(this.activityService.insert(notif.payload));
```
Comment (phenix-factory) [inline addons/mail/static/src/core/messaging_service.js:186]: This is confusing: you are deleting an insertion...

File: addons/mail/static/src/web/activity/activity_service.js [hunk 2]
```diff
@@ -53,17 +55,43 @@ export class ActivityService {
...
+            case "delete":
+                this.delete(data.payload, true);
+                break;
+        }
+    }
+
+    _serialize(activity) {
+        const data = { ...activity };
+        delete data._store;
+        return JSON.parse(JSON.stringify(data));
```
Comment (alexkuhn) [inline addons/mail/static/src/web/activity/activity_service.js:94]: Function is useless (only used once, implementation is small, etc.).
Better do this immediately where needed

File: addons/mail/static/src/web/activity/activity_service.js [hunk 3]
```diff
@@ -53,17 +55,43 @@ export class ActivityService {
...
+    _onBroadcastChannelMessage({ data }) {
+        switch (data.type) {
+            case "insert":
+                this.insert(data.payload, true);
+                break;
+            case "delete":
+                this.delete(data.payload, true);
+                break;
+        }
+    }
```
Comment (alexkuhn) [inline addons/mail/static/src/web/activity/activity_service.js:89]: Same: just use arrow function

File: addons/mail/static/src/web/activity/activity_service.js [hunk 4]
```diff
@@ -53,17 +55,43 @@ export class ActivityService {
...
+
+    _onBroadcastChannelMessage({ data }) {
+        switch (data.type) {
+            case "insert":
+                this.insert(data.payload, true);
+                break;
+            case "delete":
+                this.delete(data.payload, true);
```
Comment (alexkuhn) [inline addons/mail/static/src/web/activity/activity_service.js:86]: Not great to have positional argument as boolean.
Prefer naming them, e.g. `{ initiator: true }`

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise.

This will work for `insert/delete` (not update), which is a good coverage for now.

File: addons/mail/static/src/web/activity/activity_service.js [hunk 5]
```diff
@@ -51,24 +58,60 @@ export class ActivityService {
...
+                break;
+            case "delete":
+                this.delete(data.payload, { broadcast: false });
+                break;
+            case "reload chatter": {
+                const thread = this.env.services["mail.thread"].getThread(
+                    data.payload.resModel,
+                    data.payload.resId
+                );
+                this.env.services["mail.thread"].fetchNewMessages(thread);
```
Comment (alexkuhn) [inline addons/mail/static/src/web/activity/activity_service.js:100]: Explicitly use the service passed in constructor.
I'm doing a PR to clean them, so don't worry too much on trying to fix the problem by being inconsistent

**Pull request closed without merging**